hs628_tools/Preprocess3_module.py

```diff
@@ -0,0 +1,63 @@
+import matplotlib
+import numpy as np
+import pandas as pd
+import matplotlib
+import matplotlib.pyplot as plt
+
+
+class vizsumstat:
+
+        def __init__ (self,csvPath,colname,col_catname):
+
+                self.df = pd.read_csv(csvPath)
+                self.colname = colname
+                self.col_catname = col_catname
+                #print(self.colname)
+                self.tuple = self.coltuple(self.df, self.colname, self.col_catname)
+                self.colgroupbyclass=self.group_by(self.tuple)
+
+                
+        def group_by(self,tpls,idx=1,merge=True):
+                d = dict()
+                for tpl in tpls:
+                        k = tpl[idx]
+                        v = d.get(k,tuple()) + (tpl[:idx]+tpl[idx+1:] if merge else (tpl[:idx]+tpl[idx+1:],))
+                        d.update({k:v})
+                return d
+
+        def coltuple(self,df,col,typpe):
+                #tuplle=[]
+                for i in range(len(df[col])):
+                        for k in range(len(df[typpe])):
+                                if i==k:
+                                        #newtuple=col[i],typpe[k]
+                                        #newtuple = ['col', 'typpe'].apply(tuple, axis=1)
+                                        newtuple = list(zip(df[col], df[typpe]))
+                                        #tuplle=tuplle+newtuple
+                                #print (newtuple)
+                        return newtuple
+                
+        def summarystat(self):
+                for i in range(len(self.colgroupbyclass)):
+                       avg=np.mean(self.colgroupbyclass[i])
+                       median=np.median(  self.colgroupbyclass[i])
+                       stdev=np.std(  self.colgroupbyclass[i])
+                       plt.hist(  self.colgroupbyclass[i], normed=True, bins=20)
+                       plt.title('The "Normal" Distribution with Mean & St. Devs.')
+                       plt.xlabel("Variables"); plt.ylabel("Frequency")
```

Comment:
I am not making sense of "Variables" as the name for the x axis here.

Also, if I download a local copy of the UCI parkinson's database, saving it in the hs628_tools directory as "parkinsons_train_set.csv", and then adjust the filename below to just refer to the local copy rather than requiring your particular path structure, and then run this script specifying "HNR" and "status", then this histogram gets displayed with the first column having a frequency over 100, which suggests there is probably something wrong with your ```group_by``` method.